src/osd/OSD.h

```diff
@@ -1857,7 +1857,7 @@ class OSD : public Dispatcher,
     void dequeue_and_get_ops(PG *pg, list<OpRequestRef> *dequeued) {
       ShardData* sdata = NULL;
       assert(pg != NULL);
-      uint32_t shard_index = pg->get_pgid().ps()% shard_list.size();
+      uint32_t shard_index = pg->get_pgid().hash_to_shard(shard_list.size());
       sdata = shard_list[shard_index];
```

Comment:
I don't really like doing cleanups like this when there isn't something substantive changing (anyone working on branches that touch this code would have to deal with a silly merge conflict).  We can leave this until something substrantive changes in these methods, I think.  Please remove this commit.